Move copy of Crossgen2 tasks/targets to runtime repo and build composite image for runtime repo in parallel path

davidwrighton opened the pull request:
- See (dotnet/sdk#17540) for details of changes to tasks/targets. For any feedback about those changes please give it there.
- Place a copy of the updated tasks/targets for crossgen2 into the runtime repo
- Tweak a detail about how the shared framework sdk handles composite images
- Split the production of the bundle/installer for the runtime repo to have composite and non-composite variants

davidwrighton committed:
Move all crossgen2 tasks/targets used in Runtime repo to be local to the repo

davidwrighton committed:
Composite image production

davidwrighton committed:
Refactor how composite image creation works
- Move it to allow mixed composite and non-composite r2r in the same build, controlled by the PublishReadyToRunCompositeExclusions list
- Enhance the accuracy of how msbuild tracks which files are input vs reference for composite images
- Implement the R2R exclusion list for composite images
- Remove the always on usage of --inputbubble when compiling a composite image

davidwrighton committed:
Fix dependency tracking for composite image creation

davidwrighton committed:
Exclude the metadata and immutable collections dlls from the big composite image

davidwrighton committed:
Move ResolveReadyToRunCompilers task into the Crossgen2Tasks dll

davidwrighton committed:
Create composite bundle and package
- Note that it doesn't build properly, but hopefully that's a crossgen2 task bug

davidwrighton commented:
@dotnet/crossgen-contrib This is the runtime side PR of (dotnet/sdk#17540)
@richlander Assuming this actually works, this will produce a composite shared zip/tar.gz for use in improving container scenarios.

trylek reviewed (commented):
In general this looks awesome. I'll defer more detailed response to deeper build experts though.

AntonLapounov commented on src/installer/pkg/sfx/bundle/Microsoft.NETCore.App.Composite.Bundle.bundleproj:
```diff
@@ -0,0 +1,48 @@
+  <PropertyGroup>
+    <!--
+      Name, used to generate the bundle upgrade code. Must stay the same to allow bundles in a given
+      product band to upgrade in place.
```
This comment seems misplaced.

AntonLapounov reviewed (approved):
Thank you!

davidwrighton committed:
Fix build breaks caused by case sensitivity, pgo, and file version checks

am11 commented on eng/Versions.props:
```diff
@@ -135,6 +135,7 @@
     <XunitPerformanceApiPackageVersion>1.0.0-beta-build0015</XunitPerformanceApiPackageVersion>
     <MicrosoftDiagnosticsToolsRuntimeClientVersion>1.0.4-preview6.19326.1</MicrosoftDiagnosticsToolsRuntimeClientVersion>
     <MicrosoftDiagnosticsNETCoreClientVersion>0.2.61701</MicrosoftDiagnosticsNETCoreClientVersion>
+    <NuGetBuildTasksPackageVersion>5.10.0-preview.2.7203</NuGetBuildTasksPackageVersion>
```
Can we update line 147 to this version instead of a new entry? `RefOnlyNugetProjectModelVersion` is used here: https://github.com/dotnet/runtime/blob/28b9c72d25c3a58e85d08bc1ecf5e63a033dabe5/src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj#L45

davidwrighton committed:
- Add a scheme to perform public signing of the composite image if required
  - Required adding a new parameter to crossgen2
- Remove excess installer and package generation

mangod9 commented on src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Runtime.Composite.sfxproj:
```diff
@@ -0,0 +1,40 @@
+    <PlatformPackageType>RuntimePack</PlatformPackageType>
+    <ArchiveName>dotnet-runtime-internal-composite</ArchiveName>
+    <InstallerName Condition="'$(TargetOS)' != 'OSX'">dotnet-runtime-composite</InstallerName>
+    <InstallerName Condition="'$(TargetOS)' == 'OSX'">dotnet-runtime-composite-internal</InstallerName>
```
why does `OSX` require separate naming?

mangod9 commented on src/tasks/Crossgen2Tasks/CommonFilePulledFromSdkRepo/BuildErrorException.cs:
```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
```
should we add a comment in these `CommonFilePulledFromSdkRepo` files to indicate these are copies and any changes here should be propagated to sdk or vice-versa?

davidwrighton replied:
I've put a readme.md in the crossgen2tasks directory

davidwrighton committed:
Update based on feedback
